src/ebfm/LOOP_write_to_file.py

```diff
@@ -55,13 +57,13 @@ def main(OUTFILE, io, OUT, grid, t, time, C):
         temp_long = np.float64(OUT[varname])
 
         # Initialize TEMP storage
-        if (t - 1) % io["freqout"] == 0:
+        if t % io["freqout"] == 0:
             OUTFILE.setdefault("TEMP", {})
             OUTFILE["TEMP"][varname] = np.zeros_like(temp_long)
 
         # Handle type: sample, mean, or sum
         if var_type == "sample":
-            if (t + io["freqout"] // 2) % io["freqout"] == 0:
+            if (t + 1 + io["freqout"] // 2) % io["freqout"] == 0:  # TODO: correct?
```

Comment:
My understanding: If `io["freqout"] == 1` means that we want to write every single time step, `io["freqout"] == 2` every second time step and so on. Adding `+ io["freqout"] // 2` shifts the point where the sampling is triggered from the beginning to the middle of one period.

Example: If I set `io["freqout"] = 3` this would result in the bold time steps being written to `OUTFILE`: 0 **1** 2 | 3 **4** 5 | ... ; for even numbers are somewhat it is not really clear where the middle value is supposed to be.

Generally, I think we should clarify the expected behavior and consider moving the condition here into a function like `is_middle_t_of_every_n(...) -> bool` (open for suggestions regarding the name).